### PR TITLE
Add test_find_links

### PR DESCRIPTION
**Pull request by AnnaLangner**

### CODE REVIEW

File: script/test.py [hunk 1]
```diff
@@ -1,42 +1,27 @@
...
 
 
-def test_find_links(links):
-  for elem in links:
-    if 'href' == None:
-      assert Error in find_links      
-    else:
-      assert "All tests done."
-      
+def print_links(links):
```
Comment (klangner) [inline script/test.py]: This function is not necessary

File: script/test.py [hunk 2]
```diff
@@ -1,42 +1,27 @@
...
 
-def find_links(driver):  
-  links_elem = driver.find_elements_by_tag_name("a")    
-  return [LinkInfo(elem.text, elem.get_attribute('href')) for elem in links_elem]
-
-
-def print_link_info(links):
-  for value in links:    
-    print(value.name, value.href)
+def test_find_links(driver):
```
Comment (klangner) [inline script/test.py:31]: If you want to count number of links then:
  * Use function from the script main.py
  * At the end of the function you should expression like: `assert found_number_of_links == expected_number_of_links`
  * Rename this function to `test_link_count`

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/test.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      assert Error in lists_of_tuples      
+    else:
+      assert "All tests done."  
+  return lists_of_tuples
+
+
+def test_count_links(links):
+  found_number_of_links = len(links)
+  expected_number_of_links = 87
+  assert found_number_of_links == expected_number_of_links
```
Comment (klangner) [inline script/test.py]: It is ok, but more common is to use:
```python
assert found_number_of_links == 87
```

File: script/test.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def test_find_links(driver):
+  lists_of_tuples = find_links(driver)
+  for LinkInfo.href in lists_of_tuples:
+    if "href" == None:
+      assert Error in lists_of_tuples      
+    else:
+      assert "All tests done."  
+  return lists_of_tuples
```
Comment (klangner) [inline script/test.py]: This function shouldn't return anything.
Remember that function should do only 1 thing. In this case it is doing 2 things:
  * Testing links
  * Returning them

Also this function should have the whole test scenario. It means that it should:
 * Get the page: `driver.get("https://annalangner.github.io/Projekt-1/")`
 * find_links
 * Verify links

And Verify means comparing returned list of links to the `expected` list of links. You need to create this expected value.

File: script/test.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def test_count_links(links):
```
Comment (klangner) [inline script/test.py]: This function should be self sufficient. It means that it should:

  * Get the page: driver.get("https://annalangner.github.io/Projekt-1/")
  * find_links
  * Verify number of links

So the parameter here should be a driver not list of links

File: script/test.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+def top():  
+  driver = webdriver.Chrome(executable_path="bin/chromedriver")
+  driver.get("https://annalangner.github.io/Projekt-1/")
+  links = test_find_links(driver)
+  test_count_links(links)
+  driver.close()
+
+
+top()
```
Comment (klangner) [inline script/test.py]: Better name for this function is `main` or `test`

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/test.py (hunk 4 above)
Comment (AnnaLangner) [reply script/test.py]: I don't understand how to crate expected value? LinksInfo is not the expected value?

Comment (klangner) [reply script/test.py]: You need to create list of LinkInfo objects. Something like:
```python
expected = [LinkInfo("expected title 1", "expected href"), LinkInfo("expected title 2", "expected href")]
```

**Pull request merged**